core/src/main/java/org/elasticsearch/discovery/zen/PublishClusterStateAction.java

```diff
@@ -85,21 +88,24 @@
     private final DiscoverySettings discoverySettings;
     private final ClusterName clusterName;
     private final PendingClusterStatesQueue pendingStatesQueue;
+    private final CustomPrototypeRegistry registry;
 
     public PublishClusterStateAction(
             Settings settings,
             TransportService transportService,
             Supplier<ClusterState> clusterStateSupplier,
             NewPendingClusterStateListener listener,
             DiscoverySettings discoverySettings,
-            ClusterName clusterName) {
+            ClusterName clusterName,
+            CustomPrototypeRegistry registry) {
```

Comment:
@bleskes now that it seems that #21552 doesn't need this change, maybe we should work on adding named writable support to `XContentParser` and `DiffableValueSerializer`, so that this gets in a clean and consistent way.